### PR TITLE
Tracking of objects using the histogram-based tracker in Matlab

### LINKED ISSUE #6250: Tracking of objects using histogram based tracker in Matlab
#### Introduction
Object tracking is an application of deep learning. The program takes an initial set of object detections, develops a unique identification for each of them, and then tracks the detected objects as they move around frames in a video.
HistogramBasedTracker returns a tracker that tracks an object. It uses the histogram of pixel values to identify the tracked object.

This article will look at how to perform object tracking using the object-based tracker. We then perform tracking of the occluded objects using the Kalman filter. Also, we combine all these concepts and perform multi-object tracking. Multi-object tracking is a technique used to track multiple targets.

#### Key takeaways 
- Object tracking using the object-based tracker.
- Tracking occluded objects.
- Matlab code for tracking.
- Multi-object tracking using the same method.
#### Article quality
This article combines the idea of the histogram and the Kalman filter to perform object tracking

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by peterAdongo**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6250 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by Kayere** [comment]
Plagiarism check done ✅.

File: content/articles/tracking-of-objects-using-the-histogram-based-tracker-in-matlab/index.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+The plot for the confidence score is as follows:
+
+![Confidence score](tracking-two.png)
+
+As we can see, the tracking stops once the object is occluded. Also, the confidence scores diminishes as soon as the tracking is stopped.
+
+![Dimishing confidence scores](tracking-three.png)
+
+It is a characteristic of a histogram-based tracker. To improve this characteristic, we should create the program such that once the object is lost, the search window reinitializes when the image reappears.
+
```
Comment (M0nicah) [inline content/articles/tracking-of-objects-using-the-histogram-based-tracker-in-matlab/index.md:139]: what characteristic is mentioned here?

File: content/articles/tracking-of-objects-using-the-histogram-based-tracker-in-matlab/index.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+```Matlab
+%% Clean up
+release(vidPlayer);
+release(videoFReader);
+```
+
+If we run this code at this point, we get the following output:
+
+![Tracking](tracking-one.png)
```
Comment (M0nicah) [inline content/articles/tracking-of-objects-using-the-histogram-based-tracker-in-matlab/index.md:128]: kindly link your images using the correct format:
`/engineering-education/folder-name/image-name`

**Review by M0nicah — CHANGES_REQUESTED** [review]
hi @peterAdongo , kindly work on the suggested changes.
Also go through the article thoroughly and work on any errors that I might have missed.

Ping me when you are done.

**Comment by peterAdongo** [comment]
@M0nicah changes made.

**Review by M0nicah — APPROVED** [review]
🚀 Awesome article @peterAdongo, please review the changes I made in the commits to help improve the flow and format for your future articles.
Thanks!

**Pull request merged**